controllers/argocd/applicationset/rolebinding.go

```diff
@@ -78,32 +80,32 @@ func (asr *ApplicationSetReconciler) reconcileRoleBinding() error {
 		return nil
 	}
 
-	roleBindingChanged := false
-	fieldsToCompare := []struct {
-		existing, desired interface{}
-	}{
-		{
-			&existingRoleBinding.RoleRef,
-			&desiredRoleBinding.RoleRef,
-		},
-		{
-			&existingRoleBinding.Subjects,
-			&desiredRoleBinding.Subjects,
-		},
+	// if roleRef differs, we must delete the rolebinding as kubernetes does not allow updation of roleRef
+	if !reflect.DeepEqual(existingRoleBinding.RoleRef, desiredRoleBinding.RoleRef) {
+		asr.Logger.Info("detected drift in roleRef for rolebinding", "name", existingRoleBinding.Name, "namespace", existingRoleBinding.Namespace)
+		if err := asr.deleteRoleBinding(resourceName, asr.Instance.Namespace); err != nil {
+			return errors.Wrapf(err, "reconcileRoleBinding: unable to delete obsolete rolebinding %s", existingRoleBinding.Name)
+		}
+		return nil
+	}
```

Comment:
How will be the recreation of rolebinding handled? On next reconciliation? and does it make sense to log it as error instead of info as it is something that shouldn't have happened.

Reply:
yes @svghadi

Reply:
Should we return error in this case so that we can trigger the main `Reconcile()` loop again? My concern is that if the main `Reconcile()` loop is completed without error then the rolebinding will not be created until next reconciliation i.e by default 10hr if nothing is changed. 

Or maybe, lets merge it and figure out later when everything is in place, that it is actually  going to affect?